Automattic/jetpack
# Social: Add connection management to the admin page

Fixes https://github.com/Automattic/jetpack-reach/issues/294

## Proposed changes:

* Updated the backend so the connections return as an array, also populated the response with the needed attributes b6d9c523a1ac8ba463be3d125b35e5219629a5da
* Updated the store and frontend handlers to be able to handle the modified structure b47e440b1a9942f5f6c8774009503e8ce001dce6
* Added the new `ConnectionManagement` component ece8d5fc359723361572bdd94ece6d0276dd36ae
* Updated the `get_profile_link` function in base so it handles our new services too 312e2e4368774ca02dbb20128e9dd3d83a4000d1

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/294
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






You can switch between the new and old UI with `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );
` in your wp-config file

* Go to the social admin page
  *  You should see the connection list under the main toggle
  * All connection should have their name retrieved after the `test` endpoint finishes
  * All connection should have the correct profile picture, or the default service icon if N/A
  * Hovering over the name should get a hyperlink to the service profile page. Not working for LinkedIn
  * Goto Jetpack Settings > Sharing
  * Confirm that the Connections list is displayed
* Check that the connection toggles in the editor work as before

![CleanShot 2024-04-29 at 17 39 05 png](https://github.com/Automattic/jetpack/assets/36671565/823d8971-bf73-4457-a340-05afc2ba882d)

Jetpack settings page
<img width="1110" alt="Screenshot 2024-05-02 at 2 50 24 PM" src="https://github.com/Automattic/jetpack/assets/18226415/760174b7-a9af-4dc5-a873-91a9126ea738">

Social settings page
<img width="855" alt="Screenshot 2024-05-02 at 2 50 49 PM" src="https://github.com/Automattic/jetpack/assets/18226415/2c429feb-026b-436c-9dde-6b61b7f462fb">
